integrationtest/mapper/orm-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/orm/realbackend/multitenant/RealBackendDatabaseMultitenancyIT.java

```diff
@@ -0,0 +1,160 @@
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.realbackend.multitenant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.function.Consumer;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.search.integrationtest.mapper.orm.realbackend.testsupport.BackendConfigurations;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.GenericField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.util.common.SearchException;
+import org.hibernate.search.util.impl.integrationtest.common.FailureReportUtils;
+import org.hibernate.search.util.impl.integrationtest.mapper.orm.OrmSetupHelper;
+
+import org.junit.AssumptionViolatedException;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class RealBackendDatabaseMultitenancyIT {
+
+	public static final String TENANT_ID_1 = "TENANT 1";
+	public static final String TENANT_ID_2 = "TENANT 2";
+	public static final String TENANT_TEXT_1 = "I'm in the TENANT 1";
+	public static final String TENANT_TEXT_2 = "I'm in the TENANT 2";
+
+	@Rule
+	public OrmSetupHelper setupHelper = OrmSetupHelper.withSingleBackend( BackendConfigurations.simple() );
+
+	private SessionFactory sessionFactory;
+
+	@Test
+	public void multiTenancyStrategy_discriminator() {
+		sessionFactory = setupHelper.start()
+				.withProperty( "hibernate.search.backend.multi_tenancy.strategy", "discriminator" )
+				.withProperty( "hibernate.search.automatic_indexing.synchronization.strategy", "sync" )
+				.tenants( TENANT_ID_1, TENANT_ID_2 )
+				.setup( IndexedEntity.class );
+
+		checkMultitenancy();
+	}
+
+	@Test
+	public void multiTenancyStrategy_enabledByMapping() {
+		sessionFactory = setupHelper.start()
+				.withProperty( "hibernate.search.automatic_indexing.synchronization.strategy", "sync" )
+				.tenants( TENANT_ID_1, TENANT_ID_2 )
+				.setup( IndexedEntity.class );
+
+		checkMultitenancy();
+	}
+
+	@Test
+	public void multiTenancyStrategy_none() {
+		try {
+			setupHelper.start()
+					.withProperty( "hibernate.search.backend.multi_tenancy.strategy", "none" )
+					.tenants( TENANT_ID_1, TENANT_ID_2 )
+					.setup( IndexedEntity.class );
+		}
+		catch (AssumptionViolatedException e) {
+			// do not catch this kind of exception
+			throw e;
+		}
+		catch (Exception e) {
+			assertThat( e ).isInstanceOf( SearchException.class )
+					.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+							.typeContext( IndexedEntity.class.getName() )
+							.defaultBackendContext()
+							.failure( "Invalid backend configuration: " +
+									"index 'indexed' requires multi-tenancy but no multi-tenancy strategy is set" )
```

Comment:
I wonder why you didn't remove this check? Now that the multi-tenancy is set at the backend level, it makes no sense to pass the information "I need multi-tenancy" at the index level... The checks, if any, should really be where multi-tenancy support gets decided, i.e. at the backend level.